fda-model/structures/Inventory.cs

```diff
@@ -95,148 +109,136 @@ public static T TryGetObj<T>(object value, T defaultValue = default)
-                double beginningDamage = TryGet<double>(row[map.BeginningDamageDepth], 0);
-                int numStructures = TryGet<int>(row[map.NumberOfStructures], 1);
-                int yearInService = TryGet<int>(row[map.YearInConstruction], -999);
-                //TODO: handle number 
-                int impactAreaID = GetImpactAreaFID(point, impactAreaShapefilePath);
-                Structures.Add(new Structure(fid, point, ff_elev, val_struct, st_damcat, occtype, impactAreaID, val_cont, val_vehic, val_other, cbfips, beginningDamage, ground_elv, found_ht, yearInService, numStructures));
-
+                layer.AddAttributeColumn(nameTypePair.Item1, nameTypePair.Item2);
             }
         }
-        catch (Exception ex)
-        {
-            Console.WriteLine(ex);
-        }
-        _Occtypes = occTypes;
-    }
-    public Inventory(List<Structure> filteredStructureList, List<OccupancyType> occtypes, string impactAreaShapefilePath, string impactAreaUniqueColumnHeader)
-    {
-        Structures = filteredStructureList;
-        _Occtypes = occtypes;
-        _impactAreaSet = new PolygonFeatureLayer("Impact_Area_Set", impactAreaShapefilePath);
-        _impactAreaUniqueColumnHeader = impactAreaUniqueColumnHeader;
-    }
-    public Inventory(List<Structure> filteredStructureList, List<OccupancyType> occtypes, PolygonFeatureLayer impactAreas, string impactAreaUniqueColumnHeader)
-    {
-        Structures = filteredStructureList;
-        _Occtypes = occtypes;
-        _impactAreaSet = impactAreas;
-        _impactAreaUniqueColumnHeader = impactAreaUniqueColumnHeader;
     }
-    #endregion
     public static float[] GetGroundElevationFromTerrain(string pointShapefilePath, string TerrainPath)
     {
         PointFeatureLayer structureInventory = new PointFeatureLayer("Structure_Inventory", pointShapefilePath);
         PointMs pointMs = new PointMs(structureInventory.Points().Select(p => p.PointM()));
         TerrainLayer terrain = new TerrainLayer("Terrain", TerrainPath);
         return terrain.ComputePointElevations(pointMs);
     }
-
-    public Inventory GetInventoryTrimmmedToPolygon(int impactAreaFID)
+    private PointM ReprojectPoint(PointM point, Projection newProjection, Projection currentProjection)
     {
-        //If you don't have impact areas, just assume a single impact area, and don't filter anything. 
-        if(_impactAreaSet == null)
+        Geospatial.Vectors.Point p = Converter.Convert(point);
+        VectorExtensions.Reproject(p, currentProjection, newProjection);
+        return Converter.ConvertPtM(p);
+    }
+    private Projection GetTerrainProjection(string Pointsfilename, string terrainFilename)
+    {
+        //Check extension of terrain file
+        string extension = System.IO.Path.GetExtension(terrainFilename);
+        // If HDF, create RASTerrainLayer, then get source files. Create a GDAL Raster from any source.
+        if (extension == "hdf")
```

Comment:
Is this because if the terrain is in HDF, the file itself is not a "layer," whereas if in .tif, the .tif itself is a layer?